components/core/src/clp/GlobalMetadataDBConfig.cpp

```diff
@@ -1,110 +1,148 @@
-        }
+    if (m_metadata_db_host.empty()) {
+        throw invalid_argument("Database '--db-host' is empty.");
+    }
 
-        if (!config["name"]) {
-            throw get_yaml_missing_key_exception("name");
-        }
-        try {
-            m_metadata_db_name = config["name"].as<string>();
-        } catch (YAML::BadConversion& e) {
-            throw get_yaml_unconvertable_value_exception("name", "string");
-        }
-        if (m_metadata_db_name.empty()) {
-            throw invalid_argument("Database 'name' not specified or empty.");
-        }
+    if (cMinPort > m_metadata_db_port || cMaxPort < m_metadata_db_port) {
+        throw invalid_argument(
+                fmt::format(
+                        "Database '--db-port' is out of range [{}, {}]: {}",
+                        cMinPort,
+                        cMaxPort,
+                        m_metadata_db_port
+                )
+        );
+    }
 
-        if (!config["username"]) {
-            throw get_yaml_missing_key_exception("username");
-        }
-        try {
-            m_metadata_db_username = config["username"].as<string>();
-        } catch (YAML::BadConversion& e) {
-            throw get_yaml_unconvertable_value_exception("username", "string");
-        }
-        if (m_metadata_db_username.empty()) {
-            throw invalid_argument("Database 'username' not specified or empty.");
-        }
+    if (m_metadata_db_name.empty()) {
+        throw invalid_argument("Database '--db-name' is empty.");
+    }
 
-        if (!config["password"]) {
-            throw get_yaml_missing_key_exception("password");
-        }
-        try {
-            m_metadata_db_password = config["password"].as<string>();
-        } catch (YAML::BadConversion& e) {
-            throw get_yaml_unconvertable_value_exception("password", "string");
-        }
-        if (m_metadata_db_password.empty()) {
-            throw invalid_argument("Database 'password' not specified or empty.");
-        }
+    if (m_metadata_table_prefix.empty()) {
+        throw invalid_argument("Database '--db-table_prefix' is empty.");
+    }
 
-        if (!config["table_prefix"]) {
-            throw get_yaml_missing_key_exception("table_prefix");
-        }
-        try {
-            m_metadata_table_prefix = config["table_prefix"].as<string>();
-        } catch (YAML::BadConversion& e) {
-            throw get_yaml_unconvertable_value_exception("table_prefix", "string");
-        }
-        if (m_metadata_table_prefix.empty()) {
-            throw invalid_argument("Database 'table_prefix' not specified or empty.");
-        }
-    } else {
-        throw invalid_argument("Unknown type");
+    if (m_metadata_db_username.empty()) {
+        throw invalid_argument("Database 'CLP_DB_USER' not specified or empty.");
+    }
+
+    if (m_metadata_db_password.empty()) {
+        throw invalid_argument("Database 'CLP_DB_PASS' not specified or empty.");
```

Comment:
perhaps we should allow empty password. I would say the parameter checking is just best effort and user can't really blame us if they make mistakes themselves.